src/cart/swim/swim.c

```diff
@@ -1093,6 +1095,17 @@ swim_updates_parse(struct swim_context *ctx, swim_id_t from_id, swim_id_t id,
 	swim_ctx_lock(ctx);
 	ctx_state = swim_state_get(ctx);
 
+	rc = ctx->sc_ops->get_member_state(ctx, id, &id_state);
+	if (rc == -DER_NONEXIST || id_state.sms_status == SWIM_MEMBER_DEAD) {
```

Comment:
For the case of previous DEAD engine restarted again and update it is SWIM_MEMBER_ALIVE, I am not sure if that case will get "rc == -DER_NONEXIST || id_state.sms_status == SWIM_MEMBER_DEAD" be true?

Reply:
it's `rc == -DER_NONEXIST` in my case, but I think we should add `id_state.sms_status == SWIM_MEMBER_DEAD` too.

Reply:
For the case of DEAD engine (for example rank 6) restarting and reports itself as ALIVE, I am not sure if this PR will cause that msg be ignored? the from_id is 6 and get_member_state() will return -DER_NONEXIST() right? then will not this PR cause some trouble to rejoin it?

Reply:
If that is the case, the path will be actually much longer.

The control plane will notify the pool(not system) leader for the engine coming back, and the pool leader will send the TGT_MAP_UPDATE collective rpc that will change the node status. Long story in short, it won't go through this piece of code if I understand that correctly.

Reply:
I did some tests yesterday, it does will go through this piece of code, I see such log -
src/cart/swim/swim.c:1102 swim_updates_parse() 5: skip untrustable update from 0, rc = -1005
In my test, it the restarted rank still integrated in pool, possibly due to there are some other path propagated the ALIVE msg.
But I guess probably this PR will cause some troubles in some special case.

Reply:
I meant this code path when a rank is being added:
crt_group_primary_rank_add() -> crt_group_primary_add_internal() -> grp_add_to_membs_list() -> crt_swim_rank_add() -> crt_swim_membs_add().

Let's talk about this offline and see what's the problem you're seeing.